File: sculpt/prototype/sculpt/tuning.py
"""Phase 1 initial values for open questions (DESIGN.md §12).

All values are starting points to be answered experimentally in Phase 1.
Keep every tunable constant in this single file.
"""

GRID_SIZE = 16

LEVEL_MARGIN = {3: 16, 2: 8, 1: 4, 0: 1}

LEVEL_BLUR_BOX = {3: 8, 2: 4, 1: 2, 0: 1}

TOP_G = {3: 2, 2: 4, 1: 4, 0: 8}

COHERENCE_BONUS = 2
COHERENCE_PENALTY = 1
COHERENCE_THRESHOLD = 24

A_BAND_WIDTH = 32

DEFAULT_ITERS = {3: 2, 2: 2, 1: 2, 0: 1}

File: sculpt/src/tuning.c
#include "sculpt_tuning.h"

/* Phase 1–7 baseline values — preserved so default behavior is identical
 * to pre-Phase-8 binaries. Coherence threshold/bonus/penalty were single
 * scalars (24 / 2 / 1); they're now per-level but seeded with the same
 * value across all four levels, so existing tests don't shift.
 */
sculpt_tuning_t SCULPT_TUNING = {
    .level_margin       = { 1, 4, 8, 16 },
    .level_blur_box     = { 1, 2, 4, 8 },
    .level_top_g        = { 8, 4, 4, 2 },
    .default_iters      = { 1, 2, 2, 2 },
    .coherence_threshold = { 24, 24, 24, 24 },
    .coherence_bonus     = {  2,  2,  2,  2 },
    .coherence_penalty   = {  1,  1,  1,  1 },
    .a_band_width        = 32,
};

void sculpt_tuning_reset_default(void)
{
    sculpt_tuning_t reset = {
        .level_margin        = { 1, 4, 8, 16 },
        .level_blur_box      = { 1, 2, 4, 8 },
        .level_top_g         = { 8, 4, 4, 2 },
        .default_iters       = { 1, 2, 2, 2 },
        .coherence_threshold = { 24, 24, 24, 24 },
        .coherence_bonus     = {  2,  2,  2,  2 },
        .coherence_penalty   = {  1,  1,  1,  1 },
        .a_band_width        = 32,
    };
    SCULPT_TUNING = reset;
}
